collect_app/src/main/java/org/odk/collect/android/location/client/GoogleFusedLocationClient.java

```diff
@@ -141,7 +141,7 @@ public void stopLocationUpdates() {
     public Location getLastLocation() {
         // We need to block if the Client isn't already connected:
         if (!googleApiClient.isConnected()) {
-            googleApiClient.blockingConnect();
+            new Thread(googleApiClient::blockingConnect).start();
```

Comment:
This is a fix for another issue https://console.firebase.google.com/u/0/project/api-project-322300403941/crashlytics/app/android:org.odk.collect.android/issues/8bf1ac2f771699358e0237615316760d?time=last-seven-days&versions=v1.28.1%20(3923)&sessionEventKey=5F7C02C1035300012FE4B899DA981953_1458846445087682870

Reply:
I'm a little concerned about this. If the client isn't connected, and we spin up a new thread to connect could that cause problems for the call to `getLastLocation` (as it might happen before the client is connected)?

Also, it's likely the testing of this in `GoogleFusedLocationClientTest` (using a `timeout`) will be flakey as the call to `blockingConnect` is now async (and make the behaviour of the method indeterminate). 

Could we maybe pull this fix out of the PR and look at it as a part of a different PR?

Reply:
I tested it hardcoding `googleApiClient.disconnect();` just before `if (!googleApiClient.isConnected()) {` and it didn't cause any crashes. @lognaturel what do you think should we exclude this fix?

Reply:
If we can't block the UI thread, then this code structure doesn't makes sense. That is, like @seadowg says, if you're doing it in a different thread, that thread is blocked, but you still attempt to `getLastLocation` on the UI thread which could fail if the connection isn't instant. I think you could use `CoroutineAndWorkManagerScheduler.immediate(background, foreground)`. Splitting it off to another PR sounds best to me.

Reply:
Ok I revoked that change.